crates/sui-framework/sources/sui.move

```diff
@@ -6,18 +6,28 @@
 module sui::sui {
     use std::option;
     use sui::tx_context::TxContext;
-    use sui::balance::Supply;
+    use sui::balance::{Self, Balance};
     use sui::transfer;
     use sui::coin;
 
     friend sui::genesis;
 
+    /// The amount of Mist per Sui token based on the the fact that mist is
+    /// 10^-9 of a Sui token
+    const MIST_PER_SUI: u64 = 1000000000;
+
+    /// The total supply of Sui denominated in whole Sui tokens (10 Billion)
+    const TOTAL_SUPPLY_SUI: u64 = 10000000000;
+
+    /// The total supply of Sui denominated in Mist (10 Billion * 10^9)
+    const TOTAL_SUPPLY_MIST: u64 = 10000000000000000000;
+
     /// Name of the coin
     struct SUI has drop {}
 
     /// Register the `SUI` Coin to acquire its `Supply`.
     /// This should be called only once during genesis creation.
-    public(friend) fun new(ctx: &mut TxContext): Supply<SUI> {
+    public(friend) fun new(ctx: &mut TxContext): Balance<SUI> {
```

Comment:
A
```
assert!(tx_context::epoch_timestamp(ctx) == 0, EAlreadyMinted)
```

(assuming epoch time starts at 0) might be a good defense-in-depth mechanism here.